pkg/controller/control.go

```diff
@@ -539,27 +565,98 @@ func (c *Controller) Start(addresses ...string) error {
 	// For more details, see the following url:
 	// https://github.com/longhorn/longhorn/issues/1235
 	if !c.isUpgrade {
-		revisionCounters := make(map[string]int64)
-		for _, r := range c.replicas {
-			// The related backend is nil if the mode is ERR
-			if r.Mode == types.ERR {
-				continue
-			}
-			counter, err := c.backend.GetRevisionCounter(r.Address)
-			if err != nil {
-				return err
+		err := c.checkReplicaRevCounterSettingMatch()
+		if err != nil {
+			return err
+		}
+
+		if c.revisionCounterDisabled {
+			// Find best replica for salvge recovering, based on
+			// lastModifyTime and HeadFileSize.
+			if c.salvageRequested {
+				replicaCandidates := make(map[types.Replica]types.ReplicaSalvageInfo)
```

Comment:
Can we abstract this part as a function, e.g.,`salvageRevisionCounterDisabledReplicas()`
Similar below.

Reply:
Sure, will update. Thanks.